src/compiler/crystal/tools/context.cr

```diff
@@ -289,9 +292,10 @@ module Crystal
     private def contains_target(node)
       if loc_start = node.location
         loc_end = node.end_location || loc_start
-        # if it is not between, it could be the case that node is the top level Expressions
+        # if it is not between, it could be the case that node is the top level Expressions,
         # in which the (start) location might be in one file and the end location in another.
-        @target_location.between?(loc_start, loc_end) || loc_start.filename != loc_end.filename
+        @target_location.between?(loc_start, loc_end) || (!@in_defs && loc_start.filename != loc_end.filename)
+        # When node generated by macro and macro has yielding block, it also makes `loc_start.filename != loc_end.filename` true.
```

Comment:
Great! Can a spec be added also for this case? The specs are at `./spec/compiler/crystal/tools/context_spec.cr` and should be easy to add a sample. There are some unicode chars that act as user cursor so you can easily add specs.